instrumentation/rpc/src/main/java/brave/rpc/RpcServerHandler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.rpc;
+
+import brave.Span;
+import brave.SpanCustomizer;
+import brave.Tracer;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContext.Extractor;
+import brave.propagation.TraceContextOrSamplingFlags;
+import brave.sampler.SamplerFunction;
+
+/**
+ * This standardizes a way to instrument RPC servers, particularly in a way that encourages use of
+ * portable customizations via {@link RpcRequestParser} and {@link RpcResponseParser}.
+ *
+ * <p>Synchronous interception is the most straight forward instrumentation.
+ *
+ * <p>You generally need to:
+ * <ol>
+ *   <li>Extract any trace IDs from headers and start the span</li>
+ *   <li>Put the span in scope so things like log integration works</li>
+ *   <li>Process the request</li>
+ *   <li>Catch any errors</li>
+ *   <li>Complete the span</li>
+ * </ol>
+ * <pre>{@code
+ * RpcServerRequestWrapper wrapper = new RpcServerRequestWrapper(request);
+ * Span span = handler.handleReceive(wrapper); // 1.
+ * Result result = null;
+ * Throwable error = null;
+ * try (Scope ws = currentTraceContext.newScope(span.context())) { // 2.
+ *   return result = process(request); // 3.
+ * } catch (RuntimeException | Error e) {
+ *   error = e; // 4.
+ *   throw e;
+ * } finally {
+ *   RpcServerResponseWrapper response = result != null
```

Comment:
TODO: change this example and also client to not dance around null result (response should allow null result when there is an exception or return a constant if there is neither a result nor exception)